PhysicsTools/NanoAOD/python/taus_cff.py

```diff
@@ -111,10 +113,19 @@ def _tauIdWPMask(pattern, choices, doc="", from_raw=False, wp_thrs=None):
                                             doc="byDeepTau2018v2p5VSjet ID working points (deepTau2018v2p5)"),
 )
 
+_pnet2023 = cms.PSet(
+    decayModePNet = Var("?isTauIDAvailable('byPNetDecayMode')?tauID('byPNetDecayMode'):-1", "int16",doc="decay mode of the highest tau score of ParticleNet"),
```

Comment:
Well, it is 2 (charges) x 5 (dms) of additional floats. In addition, they are not meaning full w/o scores from other nodes (diffrent jet falvours, electrons, muons), no?
I can add an option to the "hybrid tau" producer to store all individual scores in pat:Taus (produced here on-the-fly) for specific studies if it will be useful. Those "tauIDs" can be then stored in POG/PAG/private nanoAODs/trees.

Reply:
If we succeed to come up with some central solution that fits size requirement, I think it would be more sustainable than private nano productions. For example:
- charge feducial cut is already applied, so we don't need separate scores for positive and negative taus
- to not depend on other score (vsE, mu, jet flavour) we can renormalise DM scores over the sum of tau nodes. i.e. the sum of renormalised DM stored in nano should be 1
- using reduced precision should be fine. we can even to store them as integers (while waiting for the fixed point floating support) e.g. int(renorm_dm_score * 1000).

Reply:
> If we succeed to come up with some central solution that fits size requirement, I think it would be more sustainable than private nano productions. For example:
> 
>     * charge feducial cut is already applied, so we don't need separate scores for positive and negative taus

Good point.

>     * to not depend on other score (vsE, mu, jet flavour) we can renormalise DM scores over the sum of tau nodes. i.e. the sum of renormalised DM stored in nano should be 1

Yes, this is correct, some information will be lost, but it will help to have "comparable" numbers.

>     * using reduced precision should be fine. we can even to store them as integers (while waiting for the fixed point floating support) e.g. int(renorm_dm_score * 1000).

I think it is not needed as in NanoAOD precision can be reduced using appropriate type. But it is still 5 numbers times high number of tau candidates. I can add this (hopefully today/tomorrow morning) if there are not opposite opinions.

Reply:
Regarding the space optimisation. What if we drop `tauID('byLooseCombinedIsolationDeltaBetaCorr3Hits') || (tauID('chargedIsoPtSumdR03')+max(0.,tauID('neutralIsoPtSumdR03')-0.072*tauID('puCorrPtSum'))<2.5) `? How much do we gain? I think taus that don't pass loosest WPs of neither DeepTau nor PNet, could be relevant for only very specific studies, for which a private nanos can be produced.

Reply:
I tend to agree with it, i..e. removal of simple isolation criteria from selection, but I am not sure if a PR is best place to decide this. Let's prepare this PR up to the end and then discuss changes in selection if they are mandatory.

Reply:
Requested additional information by PNet added to NanoAOD